nuleaf-source/database/events.js

```diff
@@ -28,8 +30,14 @@ exports.find = function(conditions, callback) {
     conditions = {};
   }
 
+  var skip = conditions.skip;
+  var limit = conditions.limit;
+
+  delete conditions.skip;
+  delete conditions.limit;
+
   var _conditions = buildConditions(conditions);
-  Event.find(_conditions, callback);
+  Event.find(_conditions, callback).skip(skip).limit(limit);
```

Comment:
When you pass a callback to mongooseModel.find() it will run the query immediately, therefore you can't chain skip and limit after. Please refer to mongoose documentation.